### PR TITLE
Adds the solution

### PR DESCRIPTION
**Pull request by ezmiller**
@pletcher this is ready to go pending review.

One thing you'll notice is that I didn't modularize much. For instance, I could have put the bookshelf models and stuff in a separate file, but I thought there were benefits to keeping it in one place here. Agreed?

### CODE REVIEW

File: blog_server.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    })
+    .catch((error) => {
+      console.error(error);
+      return res.sendStatus(500);
+    });
+});
+
+app.post('/user', (req, res) => {
+  if (_.isEmpty(req.body))
+    return res.sendStatus(400);
```
Comment (pletcher) [inline blog_server.js:76]: Just a style nit: I think we encourage students to use curly braces around the bodies of `if` statements; but they should understand what this means.

**Comment by pletcher** [comment]
I think in general I agree in this case (a year ago I would have disagreed, but I think that it's useful for small apps to modularize by section, not by file).

The one thing that jumps out to me is that it seems like we aren't doing much (any?) teardown in the tests just yet, which could lead to strange errors down the line. Could we add global `beforeEach` and `afterEach` hooks that wipe the database clean for each test?

File: blog_server.js (hunk 1 above)
Comment (ezmiller) [inline blog_server.js:76]: I prefer that too. My day job has the opposite in its style guide, so I always forget these days. Done.

**Comment by ezmiller** [comment]
@pletcher this should be ready to go. Tests are isolated and minor stuff fixed.

**Comment by pletcher** [comment]
:shipit: awesome!

**Pull request merged**